actioncable/lib/action_cable/connection/subscriptions.rb

```diff
@@ -46,7 +46,7 @@ def remove_subscription(subscription)
       end
 
       def perform_action(data)
-        find(data).perform_action(decode_hash(data['data']))
+        find(data).perform_action ActiveSupport::JSON.decode(data['data'])
```

Comment:
Doesn't this reintroduce the issue of how we handle JSON that has no backslashes?